pkg/pluginloader/loader.go

```diff
@@ -0,0 +1,84 @@
+package pluginloader
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+// TODO:
+// Unmarshal the YAML
+// Extract the commands using plugin.go (command structure)
+
+// Create a Cobra Command
+func createCommand(cmdStruct *CommandConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     cmdStruct.Name,
+		Short:   cmdStruct.ShortDescription,
+		Example: cmdStruct.Examples,
+		Args:    cobra.ExactArgs(len(cmdStruct.MapsTo.Args)),

```

Comment:
It is not clear for me how args mapping work here.

Reply:
at line 23
cobra.ExactArgs() takes in the number of arguments. That's what we are providing here.
and at line 25 argument mapping takes place

Reply:
Yes. This is very specific to the execution of separate processes and it will not allow us to enforce specific argument formats etc.